src/internal_rep.rs

```diff
@@ -1539,6 +1552,311 @@ fn call_to_fc_rules<'a>(
     Ok(ret)
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum FSContextType {
```

Comment:
It looks like we want the range of this later.  So the top level type should probably be a struct that holds this enum, and a range.

Reply:
I think this is obsoleted; I am able to get the range I need from the rule.